lightning/src/routing/router.rs

```diff
@@ -811,6 +811,32 @@ where L::Target: Logger {
 	// - when we want to stop looking for new paths.
 	let mut already_collected_value_msat = 0;
 
+	for (_, channels) in first_hop_targets.iter_mut() {
+		// Sort the all first_hops channels to the same node(s) in priority order of which channel
+		// we'd most like to use.
+		//
+		// First, if channels are below `recommended_value_msat`, sort them in descending order,
+		// preferring larger channels to avoid splitting the payment into more MPP parts than is
+		// required.
+		//
+		// Second, because simply always sorting in descending order would always use our largest
+		// available outbound capacity, needlessly fragmenting our available channel capacities,
+		// sort channels above `recommended_value_msat` in ascending order, preferring channels
+		// which have enough, but not too much, capacity for the payment.
+		channels.sort_unstable_by(|chan_a, chan_b| {
+			if chan_b.outbound_capacity_msat >= recommended_value_msat && chan_a.outbound_capacity_msat >= recommended_value_msat {
+				// Sort in ascending order
+				chan_a.outbound_capacity_msat.cmp(&chan_b.outbound_capacity_msat)
+			} else if chan_a.outbound_capacity_msat > recommended_value_msat {
+				core::cmp::Ordering::Less
+			} else if chan_b.outbound_capacity_msat > recommended_value_msat {
+				core::cmp::Ordering::Greater
```

Comment:
Just some potential cleanup I noticed when looking through this, that would improve readability. Aren't these else if clauses unnecessary? I can't think of a scenario where they would deliver a different result than the final else clause